### DISCUSSION TITLE
RPC with empty types?

### DISCUSSION THREAD

**Discussion by rodolphoasb** [honojs/hono#2432] [Q&A] [answered]
Hey, I'm new to Hono and trying to figure out why this doesn't have the output with type: `{message: string}`

![image](https://github.com/honojs/hono/assets/8127858/cebb1f78-e51b-485b-b8e0-adf421654ac1)
![image](https://github.com/honojs/hono/assets/8127858/290e0164-da0f-4f34-801d-f61833d69032)

**Comment by yusukebe** [comment]
Hi @rodolphoasb 

Hmm. That's not reproduced. If you create a minimal project to reproduce it, we can take a look.

<img width="616" alt="Screenshot 2024-03-29 at 5 12 23" src="https://github.com/honojs/hono/assets/10682/8ff379a6-fe86-46f5-8449-bb5f7b22b134">

**Reply by rodolphoasb** [reply]
Hey @yusukebe, hope you're doing well!

Here's the repo: https://github.com/rodolphoasb/remix-monorepo

**Reply by yusukebe** [reply]
Hmm. Weird. It works well on my machine.

<img width="533" alt="Screenshot 2024-03-29 at 6 24 30" src="https://github.com/honojs/hono/assets/10682/952ad7e4-48ac-434f-abdc-6bc4c9170b27">

How about trying to remove `node_modules` and re-install dependencies?

**Reply by rodolphoasb** [reply]
I just tried and still no luck, super weird... 😕

**Reply by rodolphoasb** [reply]
Here's my system info:

```
  System:
    OS: macOS 13.3.1
    CPU: (8) arm64 Apple M1
    Memory: 100.42 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.10.0 - ~/.nvm/versions/node/v20.10.0/bin/node
    Yarn: 1.22.15 - /opt/homebrew/bin/yarn
    npm: 10.2.3 - ~/.nvm/versions/node/v20.10.0/bin/npm
    pnpm: 8.15.5 - ~/.nvm/versions/node/v20.10.0/bin/pnpm
    bun: 1.0.3 - ~/.bun/bin/bun
    Watchman: 2024.01.22.00 - /opt/homebrew/bin/watchman
  Browsers:
    Chrome: 123.0.6312.87
    Safari: 16.4
```

**Reply by yusukebe** [reply]
Hmmmm... Sometimes restarting the IDE (VS Code) will make it work, though.

**Reply by rodolphoasb** [reply]
It looks like I can also repro it with Stackblitz:

![image](https://github.com/honojs/hono/assets/8127858/bda96b0f-f695-44b0-83cc-ceda61c4fa2f)

**Reply by yusukebe** [reply, marked as answer]
@rodolphoasb 

Try to use this `tsconfig.json`:

```ts
{
  "compilerOptions": {
    "strict": true,
    "target": "ES2022",
    "module": "ES2022",
    "baseUrl": ".",
    "moduleResolution": "Bundler",
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,
    "skipLibCheck": true,
    "jsx": "react-jsx",
    "jsxImportSource": "hono/jsx",
  },
  "include": ["src"]
}
```

**Reply by rodolphoasb** [reply]
Awesome, that worked! Thank you!
